### PR TITLE
introducing automatic english doc generation...

### PR DESCRIPTION
**Pull request by peuter**
... from source code comments for plugins and widgets. Add the basic doc structure and add some autogenerated content.

### CODE REVIEW

File: doc/manual/en/config/widgets/plugins/openweathermap/index.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+=========================
+
+.. api-doc:: openweathermap
+
+Description
+-----------
+
+.. ###START-WIDGET-DESCRIPTION### Please do not change the following content. Changes will be overwritten
+
+structure_plugin.js (c) 2015 by Stefan Borchert (stefan@borchert.cc)
```
Comment (ChristianMayer) [inline doc/manual/en/config/widgets/plugins/openweathermap/index.rst]: I think the auto import made a mistake here

File: doc/manual/en/config/widgets/plugins/strftime/index.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+.. api-doc:: strftime
+
+Description
+-----------
+
+.. ###START-WIDGET-DESCRIPTION### Please do not change the following content. Changes will be overwritten
+
+This plugins integrates formated date and clock strings into based on strftime.
+
+Thanks to Michael Markstaller for implementing the jqclock plugin as reference.
```
Comment (ChristianMayer) [inline doc/manual/en/config/widgets/plugins/strftime/index.rst]: I don't think that this line belongs in a manual.

When we take that line out of the source file would that automatically be fixed here or would that require a manual process call?

File: doc/manual/en/config/widgets/unknown/index.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+.. api-doc:: Unknown
+
+Description
+-----------
+
+.. ###START-WIDGET-DESCRIPTION### Please do not change the following content. Changes will be overwritten
+
+Fallback widget shown when an unknown widget is defined in the configuration.
+You must not use this one directly.
```
Comment (ChristianMayer) [inline doc/manual/en/config/widgets/unknown/index.rst]: Can we suppress the docu generation for this widget?

It's just an error fallback.

File: src/editor/lib/DocumentationMapping.json [hunk 4]
```diff
@@ -1,40 +1,4 @@
 {
   "_base": "http://test.cometvisu.org/CometVisu/", 
-  "audio": "/latest/manual/config/widgets/audio/index.html#audio",
```
Comment (ChristianMayer) [inline src/editor/lib/DocumentationMapping.json]: Why are all these lines deleted?

File: doc/manual/en/config/widgets/plugins/strftime/index.rst (hunk 2 above)
Comment (peuter) [inline doc/manual/en/config/widgets/plugins/strftime/index.rst]: Actually the content between `.. ###START-WIDGET-DESCRIPTION###` and `.. ###END-WIDGET-DESCRIPTION###` gets replaced on every build with the content taken from the source. So removing that from source should be fine.

File: src/editor/lib/DocumentationMapping.json (hunk 4 above)
Comment (peuter) [inline src/editor/lib/DocumentationMapping.json]: By accident, didn't meant to commit this change. Will revert that.

File: src/plugins/gauge/structure_plugin.js [hunk 5]
```diff
@@ -16,29 +16,7 @@
  */
  
 /**
- * This plugins integrates Gauge (in canavas) into the visualization
- *
- *
- * short documentation
- *
- * attributes:
```
Comment (peuter) [inline src/plugins/gauge/structure_plugin.js]: This information is totally outdated, thats why I removed it.

**Comment by peuter** [comment]
See commits ;-)

**Pull request merged**